src/xpcc/driver/can/mcp2515_impl.hpp

```diff
@@ -423,3 +437,17 @@ xpcc::Mcp2515<SPI, CS, INT>::readIdentifier(uint32_t& identifier)
 		return false;
 	}
 }
+
+xpcc::Abandonment can_assertion_handler(char *module, char *, char *, uintptr_t)
+{
+    if (!strncmp(module,   "mcp2515",  7) and
+    	!strncmp(location, "init",     4) and
+    	!strncmp(failure,  "readback", 8))
+    {
+        return xpcc::Abandonment::Fail;
+    }
+    return xpcc::Abandonment::DontCare;
+}
+
+// Register assertion handler with system
+XPCC_ASSERTION_HANDLER(can_assertion_handler);
```

Comment:
> And when he/she forgets to define one?

The default behavior is to abandon the program execution. This is the de-facto behavior when failing an `assert`. However, this abandonment can be overwritten by the user via an assertion handler.

> How does she/he know about the necessity to define one?

Through documentation. There should be a description of assertions somewhere™, probably in the function's doxygen. The idea was that you could scan the source code for `xpcc_assert` and the containing function and autogenerate this documentation. But I didn't get to that yet.

> Any chance to get a weak default handler?

I'm confused about what you want (presumably aggregated by poor documentation on my part). The assertion handler above _is_ a "strong default handler", cos you `Abandonment::Fail` if this assertion is triggered! If you wanted a "weak" one, you would have to return `Abandonment::Ignore` (which is overruled by any assertion handler returing `Abandonment::Fail`).

But why would you want to do that? Your previous commit with the `XPCC_LOG_ERROR` made it seem like this is an irrecoverable error, hence the `xpcc_assert`. If this is not the case, and you can recover from the error, return an error code.
There's also `xpcc_assert_debug()` which is enabled in debug builds (which we don't have, so it's currently never enabled ;-). 

Both `xpcc_assert` and `xpcc_assert_debug` return the condition, so you can do stuff like this:
```cpp
if (not xpcc_assert(test_condition, ...))
{
    // user chose to ignore assertion
    // lets log this failure, but continue anyways
    XPCC_LOG_ERROR << "foobar failed, system degraded, implosion imminent" << xpcc::endl;
}
```

> Hmm, only a mother could love that?

Yes, of course. _And_ I wrote an extensive blog post about the research and implementation… double bonus 😛